test/Libraries/Microsoft.Extensions.AI.Integration.Tests/PromptBasedFunctionCallingChatClient.cs

```diff
@@ -181,14 +183,14 @@ answer the user's question without repeating the same tool call.
     {
         Name = tool.Metadata.Name,
         Description = tool.Metadata.Description,
-        Arguments = tool.Metadata.Parameters.ToDictionary(
-            p => p.Name,
+        Arguments = tool.Metadata.UnderlyingMethod!.GetParameters().ToDictionary(
```

Comment:
What guarantees UnderlyingMethod is non-null here? If this is processing arbitrary AIFunctions coming in from Tools, couldn't this be null? Same question about the Name below.

Reply:
It's test code hitting AIFunctions that we define. I could coalesce the null case if you want.

Reply:
It's not really a test. It's a sample for writing a custom IChatClient middleware component.